Source/EasyNetQ.Tests/AutoSubscriberTests/When_auto_subscribing.cs

```diff
@@ -20,7 +22,10 @@ public When_auto_subscribing()
     {
         mockBuilder = new MockBuilder();
 
-        var autoSubscriber = new AutoSubscriber(mockBuilder.Bus, "my_app");
+        var services = new ServiceCollection();
+        serviceProvider = services.BuildServiceProvider();
+
+        var autoSubscriber = new AutoSubscriber(mockBuilder.Bus, serviceProvider, "my_app");
```

Comment:
@Atallah0 @zidad 

Let's decide what to do with AutoSubscriber. If it is integrated with DI, we could consider doing two alternatives:
1. Do not move it out from this assembly and register it in DI containter as well in RegisterDefaultServices, so that it can be resolved in background service and started/stopped.
2. Move it right now, but we need to discuss how to do that. We might be need something else as alternative to it.

Reply:
it should work now, so let's decide what to do with it later and merge this PR